nuxeo-services/nuxeo-platform-dublincore/src/test/java/org/nuxeo/ecm/platform/dublincore/listener/DublinCoreListenerTest.java

```diff
@@ -0,0 +1,250 @@
+        ctx.setProperties(properties);
+
+        eventService.fireEvent(ctx.newEvent(ABOUT_TO_CREATE));
+
+        verifyZeroInteractions(storageService);
+    }
+
+    @Test
+    public void shouldDoNothingWhenDocumentIsVersion() {
+        DocumentModel doc = Mockito.mock(DocumentModel.class);
+        when(doc.isVersion()).thenReturn(true);
+
+        DocumentEventContext ctx = new DocumentEventContext(session, session.getPrincipal(), doc);
+        eventService.fireEvent(ctx.newEvent(ABOUT_TO_CREATE));
+
+        verifyZeroInteractions(storageService);
+    }
+
+    @Test
+    public void shouldDoNothingWhenDocumentHasSystemDocumentFacet() {
+        DocumentModel doc = Mockito.mock(DocumentModel.class);
+        when(doc.hasFacet("SystemDocument")).thenReturn(true);
+
+        DocumentEventContext ctx = new DocumentEventContext(session, session.getPrincipal(), doc);
+        eventService.fireEvent(ctx.newEvent(ABOUT_TO_CREATE));
+
+        verifyZeroInteractions(storageService);
+    }
+
+    @Test
+    public void shouldDoNothingWhenDocumentIsProxyAndEventIsAboutToCreate() {
+        DocumentModel doc = Mockito.mock(DocumentModel.class);
+        when(doc.isProxy()).thenReturn(true);
+
+        DocumentEventContext ctx = new DocumentEventContext(session, session.getPrincipal(), doc);
+        eventService.fireEvent(ctx.newEvent(ABOUT_TO_CREATE));
+
+        verifyZeroInteractions(storageService);
+    }
+
+    @Test
+    public void shouldDoNothingWhenDocumentIsProxyAndIsImmutable() {
+        DocumentModel doc = Mockito.mock(DocumentModel.class);
+        when(doc.isProxy()).thenReturn(true);
+        when(doc.isImmutable()).thenReturn(true);
+
+        DocumentEventContext ctx = new DocumentEventContext(session, session.getPrincipal(), doc);
+        eventService.fireEvent(ctx.newEvent(ABOUT_TO_CREATE));
+
+        verifyZeroInteractions(storageService);
+    }
+
+    @Test
+    public void shouldInvokeStorageServiceWhenEventIsAboutToCreate() {
+        DocumentModel doc = session.createDocumentModel("File");
+
+        DocumentEventContext ctx = new DocumentEventContext(session, session.getPrincipal(), doc);
+        Event event = ctx.newEvent(ABOUT_TO_CREATE);
+        eventService.fireEvent(event);
+
+        Calendar expectedDate = Calendar.getInstance();
+        expectedDate.setTime(new Date(event.getTime()));
+
+        verify(storageService).setCreationDate(eq(doc), eq(expectedDate));
+        verify(storageService).setModificationDate(eq(doc), eq(expectedDate));
+        verify(storageService).addContributor(eq(doc), eq(event));
+    }
+
+    @Test
+    public void shouldInvokeStorageServiceButNotSetCreationDateWhenEventIsBeforeUpdateAndDocumentIsDirty() {
+        DocumentModel doc = Mockito.mock(DocumentModel.class);
+        when(doc.isDirty()).thenReturn(true);
+
+        Map<String, Serializable> properties = new HashMap<>();
+        properties.put(DOCUMENT_DIRTY, true);
```

Comment:
You can use `Collections.singletonMap`.

Reply:
I tried that approach, but it seems that is other underneath service that adds more properties.
With the singletonMap an exception will be thrown.